workos/utils/connection_types.py

```diff
@@ -1,48 +1,39 @@
-from enum import Enum
+from typing import Literal
 
 
-class ConnectionType(Enum):
-    ADFSSAML = "ADFSSAML"
-    AdpOidc = "AdpOidc"
-    AppleOAuth = "AppleOAuth"
-    Auth0SAML = "Auth0SAML"
-    AzureSAML = "AzureSAML"
-    CasSAML = "CasSAML"
-    CloudflareSAML = "CloudflareSAML"
-    ClassLinkSAML = "ClassLinkSAML"
-    CyberArkSAML = "CyberArkSAML"
-    DuoSAML = "DuoSAML"
-    GenericOIDC = "GenericOIDC"
-    GenericSAML = "GenericSAML"
-    GitHubOAuth = "GitHubOAuth"
-    GoogleOAuth = "GoogleOAuth"
-    GoogleSAML = "GoogleSAML"
-    JumpCloudSAML = "JumpCloudSAML"
-    KeycloakSAML = "KeycloakSAML"
-    LastPassSAML = "LastPassSAML"
-    MagicLink = "MagicLink"
-    MicrosoftOAuth = "MicrosoftOAuth"
-    MiniOrangeSAML = "MiniOrangeSAML"
-    NetIqSAML = "NetIqSAML"
-    OktaSAML = "OktaSAML"
-    OneLoginSAML = "OneLoginSAML"
-    OracleSAML = "OracleSAML"
-    PingFederateSAML = "PingFederateSAML"
-    PingOneSAML = "PingOneSAML"
-    RipplingSAML = "RipplingSAML"
-    SalesforceSAML = "SalesforceSAML"
-    ShibbolethGenericSAML = "ShibbolethGenericSAML"
-    ShibbolethSAML = "ShibbolethSAML"
-    SimpleSamlPhpSAML = "SimpleSamlPhpSAML"
-    VMwareSAML = "VMwareSAML"
-
-    @classmethod
-    def providers(cls):
-        """Returns a generator of all connection types/providers.
-        This is only needed as a workaround for providers passed
-        as a string connection type.
-
-        Returns:
-            generator(list): A lazy list of all connection types
-        """
-        return (connection_type.value for connection_type in ConnectionType)
+ConnectionType = Literal[
+    "ADFSSAML",
+    "AdpOidc",
+    "AppleOAuth",
+    "Auth0SAML",
+    "AzureSAML",
+    "CasSAML",
+    "CloudflareSAML",
+    "ClassLinkSAML",
+    "CyberArkSAML",
+    "DuoSAML",
+    "GenericOIDC",
+    "GenericSAML",
+    "GitHubOAuth",
+    "GoogleOAuth",
+    "GoogleSAML",
+    "JumpCloudSAML",
+    "KeycloakSAML",
+    "LastPassSAML",
+    "LoginGovOidc",
+    "MagicLink",
+    "MicrosoftOAuth",
+    "MiniOrangeSAML",
+    "NetIqSAML",
+    "OktaSAML",
+    "OneLoginSAML",
+    "OracleSAML",
+    "PingFederateSAML",
+    "PingOneSAML",
+    "RipplingSAML",
+    "SalesforceSAML",
+    "ShibbolethGenericSAML",
+    "ShibbolethSAML",
+    "SimpleSamlPhpSAML",
+    "VMwareSAML",
+]
```

Comment:
The API types also include [`pending`](https://github.com/workos/workos/blob/a91712330d5e55492c79ef71c480eb27497d750b/packages/api-json-models/src/shared/connection-type.ts#L7) and [`Auth0Migration`](https://github.com/workos/workos/blob/a91712330d5e55492c79ef71c480eb27497d750b/packages/api-json-models/src/shared/connection-type.ts#L12) types.

I think whoever owns SSO (is that us? LOL) could give us guidance around whether it's safe to omit these.